source/common/api/win32/os_sys_calls_impl.cc

```diff
@@ -364,5 +364,35 @@ SysCallSocketResult OsSysCallsImpl::accept(os_fd_t sockfd, sockaddr* addr, sockl
   return {rc, 0};
 }
 
+SysCallBoolResult OsSysCallsImpl::socketTcpInfo([[maybe_unused]] os_fd_t sockfd,
+                                                [[maybe_unused]] tcp_info* tcp_info) {
+#ifdef SIO_TCP_INFO
+  TCP_INFO_v0 win_tcpinfo;
+  DWORD infoVersion = 0;
+  DWORD bytesReturned = 0;
+  int rc = ::WSAIoctl(sockfd, SIO_TCP_INFO, &infoVersion, sizeof(infoVersion), &win_tcpinfo,
+                      sizeof(win_tcpinfo), &bytesReturned, nullptr, nullptr);
+  if (rc == SOCKET_ERROR) {
+    return {false, ::WSAGetLastError()};
+  } else {
```

Comment:
nit: not big deal, but you could get rid of the else case

Reply:
fixed in 7c9137b7a3fb86ccae63cbd4018683cc47b7cb4d